aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -1094,7 +1094,19 @@ ReduceConfig setReduceConfig(const TensorIterator& iter){
   constexpr int min_values_per_thread = 16;
   constexpr int max_values_per_thread = 256;
 
-  if (config.values_per_thread() >= block_height * 16 || config.values_per_thread() >= max_values_per_thread) {
+  const int warp_split_threshold =
+      std::min<int>(block_height * 16, max_values_per_thread);
+  const int num_mp =
+      at::cuda::getCurrentDeviceProperties()->multiProcessorCount;
+  bool force_splitting_output = false;
```

Comment:
Would it make sense to just `#ifdef` different ROCM/non-ROCM conditions rather than only computing this predicate on ROCM?

Reply:
@eqy 
Can this be approved ?

Reply:
It looks like this introduces `force_splitting_output` which is always `false` for the non-ROCM case. Would it make sense to move it into the ROCM-only ifdef as well?

Reply:
The variable needs to be defined in both cases because it is used in the following if statement so it cannot exist just in the ROCM case.

Reply:
How is it used? It appears to be added in this PR and could be set to a constant in the non-ROCM case.

Reply:
It's used in the condition of the if statement that follows.